charts/covidDataExplorer/CovidCountryPicker.tsx

```diff
@@ -90,6 +90,7 @@ export class CountryPicker extends React.Component<{
 
     set metric(metric) {
         this.props.covidDataExplorer.props.params.countryPickerMetric = metric
+        this.sortOrder = metric === "location" ? SortOrder.asc : SortOrder.desc
```

Comment:
Should this be a computed? What happens if someone changes the sort order?

Reply:
I think I would put this in the `onChange` handler, where the `metric` setter is also called: 

https://github.com/owid/owid-grapher/blob/62185f015e8fff2919a1ada77dc2500aba50f3d7/charts/covidDataExplorer/CovidCountryPicker.tsx#L402-L408

This is just because I try to avoid setters with side effects (though I have probably written some in the past).

A concrete example: Maybe sometime down the line there's this implementation:

```js
this.sortOrder = ...
this.metric = ...
```

Now the order in which you assign these matters, and it's not obvious.

Reply:
@breck7 not sure if it should be computed, as we need an observable that responds to user interaction. The code should just run if the user changes the metric and sets the "default" sort state. `sortOrder` can also be modified by the user clicking the change sort button.

@danielgavrilov Good idea, I moved it into the onChange handler. I'm not sure what the concern with the order is -- as I use the value of the new metric rather than state. However, if you're referring to potential inconsistent state, I also wrapped it in a runInAction transaction.